src/VirtualClient/VirtualClient.Actions/SPECview/SpecViewExecutor.cs

```diff
@@ -0,0 +1,203 @@
+                    string[] subdirectories = this.fileSystem.Directory.GetDirectories(this.Package.Path, "results_*", SearchOption.TopDirectoryOnly);
+
+                    // Sort the "results_" subdirectories by creation time in descending order and take the first one
+                    string resultsFileDir = subdirectories.OrderByDescending(d => this.fileSystem.Directory.GetCreationTime(d)).FirstOrDefault();
+                    string resultsFilePath = this.PlatformSpecifics.Combine(resultsFileDir, "resultCSV.csv");
+                    string resultsContent = this.fileSystem.File.ReadAllText(resultsFilePath);
+
+                    SpecViewMetricsParser resultsParser = new (resultsContent);
+                    IList<Metric> metrics = resultsParser.Parse();
+
+                    this.MetadataContract.AddForScenario(
+                           this.Scenario,
+                           workloadProcess.FullCommand(),
+                           toolVersion: "2020 v3.0");
+                    this.MetadataContract.Apply(telemetryContext);
+
+                    this.Logger.LogMetrics(
+                        "SPECview",
+                        this.Scenario,
+                        workloadProcess.StartTime,
+                        workloadProcess.ExitTime,
+                        metrics,
+                        null,
+                        commandArguments,
+                        this.Tags,
+                        telemetryContext);
+                }
+                catch (SchemaException exc)
+                {
+                    EventContext relatedContext = telemetryContext.Clone()
+                        .AddError(exc);
+
+                    this.Logger.LogMessage($"{nameof(SpecViewExecutor)}.WorkloadOutputParsingFailed", LogLevel.Warning, relatedContext);
+                }
+                catch (FileNotFoundException exc)
+                {
+                    EventContext relatedContext = telemetryContext.Clone()
+                        .AddError(exc);
+
+                    this.Logger.LogMessage($"{nameof(SpecViewExecutor)}.WorkloadOutputFileNotFound", LogLevel.Warning, relatedContext);
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Run the Spec View Workload
+        /// </summary>
+        private Task ExecuteWorkloadAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            IList<Metric> metrics = new List<Metric>();
+
+            string commandArguments = this.CommandArguments;
+
+            EventContext relatedContext = telemetryContext.Clone()
+                .AddContext("executable", this.ExecutablePath)
+                .AddContext("commandArguments", commandArguments);
+
+            return this.Logger.LogMessageAsync($"{nameof(SpecViewExecutor)}.ExecuteWorkload", relatedContext, async () =>
```

Comment:
We should now have an extension methods on VCComponent. this.ExecuteCommandAsync that has what this methods does, let me know if it works for you